src/main/java/com/intellij/compiler/notNullVerification/AnnotationThrowOnNullMethodVisitor.java

```diff
@@ -68,7 +68,9 @@ public AnnotationVisitor visitAnnotation(final String annotation, final boolean
     @Override
     @NotNull
     protected String getThrowMessage(int notNullParam) {
-        return "Argument " + getSourceCodeParameterNumber(notNullParam) + " for @NotNull parameter of " + className + "." + methodName + " must not be null";
+        int pnum = getSourceCodeParameterNumber(notNullParam);
+        String pname = parameterNames == null ? "" : " '" + parameterNames.get(pnum) + "'";
```

Comment:
theoretically, it could. in this case, it will fail at bytecode instrumentation phase. i can easily add an extra condition here to fall back to the nameless message.
i'd probably wait for this to actually happen, though - that would be an indication of a flaw in the overall logic of the approach.